lib/types.js

```diff
@@ -41,10 +41,65 @@ const exclude = function (types, excludes) {
   return typesCopy;
 }
 
+const excludeImmutable = function (types, excludes) {
+  if (!excludes || excludes.length < 1) {
+    return types;
+  }
+
+  let mutations = [];
+
+  for (let typeIndex = 0; typeIndex < types.length; typeIndex++) {
+    const doc = types[typeIndex];
+    for (let defIndex = 0; defIndex < doc.definitions.length; defIndex++) {
+      const def = doc.definitions[defIndex];
+      if (def.kind !== 'ObjectTypeDefinition') {
+        continue;
+      }
+      if (['Query', 'Mutation', 'Subscription'].indexOf(def.name.value) > -1) {
+        let fields = def.fields.filter((field) => {
+          if (check(def.name.value, field.name.value, excludes)) {
+            debug(`excluding ${def.name.value}.${field.name.value} from import`);
+            return false;
+          }
+          return true;
+        });
+
+        if (fields.length !== def.fields.length) {
+          mutations.push({
+            typeIndex,
+            defIndex,
+            fields
+          });
+        }
+      }
+    }
+  }
+
+  for (const mutation of mutations) {
+    const doc = types[mutation.typeIndex];
+    const definition = doc.definitions[mutation.defIndex];
+
+    const newDefinitions = [
+      ...doc.definitions
+    ];
+    newDefinitions[mutation.defIndex] = {
+      ...definition,
+      fields: mutation.fields
+    };
+
+    types[mutation.typeIndex] = {
+      ...doc,
+      definitions: newDefinitions
+    };
+  }
+
+  return types;
+}
+
 const getImportedTypes = function (component, excludes) {
   const types = component._types.map((type) => Gql`${type}`);
   const importedTypes = component._importedTypes;
-  return exclude([...types, ...importedTypes], excludes);
+  return excludeImmutable([...types, ...importedTypes], excludes);
```

Comment:
I imagine the `exclude` function is dead code then? Why not just remove it.

Reply:
yea, i don't intend for both of these to exist; i just wanted to have both versions up at the same time for comparison and depending on which one we want to move forward with, that'll become the new `exclude`